### PR TITLE
Support 1.20.4

### PR DESCRIPTION
**Pull request by rom1504**

### CODE REVIEW

**Comment by extremeheat** [comment]
There was an issue with chat command params fixed with https://github.com/PrismarineJS/minecraft-data/pull/850
Currently failing on this error
```js
     Error: Error: crafting: non craftingTable used as craftingTable: minecraft:enchantment
      at EventEmitter.craft (lib/plugins/craft.js:32:13)
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
      at async /workspaces/mineflayer/test/externalTests/crafting.js:57:3
```
This might be fixed by @wgaylord's recipes PR in https://github.com/PrismarineJS/minecraft-data/pull/843

I added some extra context to the error messages and swapped Node.js once() with a mineflayer one with a default 20s timeout (only place that needed unlimited time seems to be bot.flyTo()) to make errors more obvious. Seems that flakiness in some of the tests seems to be caused by windows not opening: https://github.com/PrismarineJS/mineflayer/actions/runs/7999230618/job/21846700577?pr=3310

```js
  1) mineflayer_external 1.13.2v
       useChests:
     Error: Event windowOpen did not fire within timeout of 20000ms
      at onceWithCleanup (lib/promise_utils.js:62:26)
      at once (lib/promise_utils.js:76:10)
      at EventEmitter.openBlock (lib/plugins/inventory.js:366:28)
      at EventEmitter.openContainer (lib/plugins/chest.js:19:25)
      at withdrawBones (test/externalTests/useChests.js:54:29)
      at /home/runner/work/mineflayer/mineflayer/test/externalTests/useChests.js:119:9
      at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
```

**Comment by wgaylord** [comment]
> 
> ```js
>      Error: Error: crafting: non craftingTable used as craftingTable: minecraft:enchantment
>       at EventEmitter.craft (lib/plugins/craft.js:32:13)
>       at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
>       at async /workspaces/mineflayer/test/externalTests/crafting.js:57:3
> ```
> 
> This might be fixed by @wgaylord's recipes PR in [PrismarineJS/minecraft-data#843](https://github.com/PrismarineJS/minecraft-data/pull/843)

After a bit of poking around this looks more like an issue in https://github.com/PrismarineJS/prismarine-windows/blob/master/index.js then with the recipes stuff itself. (I didn't actually read the error well when I suggested that my PR to fix the recipes may fix it)    Looks like for some reason it thinks it opened an enchantment table instead of a crafting table.

**Comment by wgaylord** [comment]
This should fix the crafting issue. https://github.com/PrismarineJS/prismarine-windows/pull/105

**Comment by Lozarth** [comment]
```
2024-02-23T06:26:42.7567449Z   1 failing
2024-02-23T06:26:42.7567669Z 
2024-02-23T06:26:42.7570657Z   1) mineflayer_external 1.17.1v
2024-02-23T06:26:42.7571226Z        exampleBlockFinder:
2024-02-23T06:26:42.7571772Z      Error: Promise timed out.
2024-02-23T06:26:42.7574286Z       at /home/runner/work/mineflayer/mineflayer/lib/promise_utils.js:83:13
2024-02-23T06:26:42.7575604Z       at async Object.runExample (test/externalTests/plugins/testCommon.js:211:7)
2024-02-23T06:26:42.7576683Z       at async /home/runner/work/mineflayer/mineflayer/test/externalTests/exampleBlockFinder.js:4:3
2024-02-23T06:26:42.7577303Z 
2024-02-23T06:26:42.7577311Z 
2024-02-23T06:26:42.7577318Z 
2024-02-23T06:26:42.7897291Z ##[error]Process completed with exit code 1.
```

How can I help contribute and fix this error?

File: lib/plugins/chat.js [hunk 1]
```diff
@@ -107,8 +107,12 @@ function inject (bot, options) {
   })
 
   addDefaultPatterns()
+  bot._client.on('player_chat', (msg) => console.dir(msg, { depth: null }))
+  bot._client.on('profileless_chat', (msg) => console.dir(msg, { depth: null }))
 
   bot._client.on('playerChat', (data) => {
+    console.log('playerChat')
```
Comment (rom1504) [inline lib/plugins/chat.js:114]: please remove

File: lib/plugins/chat.js [hunk 2]
```diff
@@ -107,8 +107,12 @@ function inject (bot, options) {
   })
 
   addDefaultPatterns()
+  bot._client.on('player_chat', (msg) => console.dir(msg, { depth: null }))
```
Comment (rom1504) [inline lib/plugins/chat.js:110]: debug code

**Comment by 3jq** [comment]
hoorah when though?

**Pull request merged**